File: 0682-baseball-game/0682-baseball-game.js
/**
 * @param {string[]} operations
 * @return {number}
 */
var calPoints = function(operations) {
    const stack = [];

    operations.forEach(op => {
        if (op === '+') {
            stack.push(stack[stack.length - 1] + stack[stack.length - 2]);
        } else if (op === 'D') {
            stack.push(2 * stack[stack.length - 1]);
        } else if (op === 'C') {
            stack.pop();
        } else {
            stack.push(parseInt(op));
        }
    });

    return stack.reduce((acc, score) => acc + score, 0);
};

File: 0682-baseball-game/0682-baseball-game.java
class Solution {
    public int calPoints(String[] operations) {
        Stack<Integer> stack = new Stack<>();
        
        Arrays.stream(operations).forEach(op ->{
            if (op.equals("+")) {
                int top = stack.pop();
                int newTop = top + stack.peek();
                stack.push(top);
                stack.push(newTop);
            } else if (op.equals("D")) {
                stack.push(2 * stack.peek());
            } else if (op.equals("C")) {
                stack.pop();
            } else {
                stack.push(Integer.parseInt(op));
            }
        });
        
        int res = 0;
        for(int val : stack){
            res += val;
        }
        return res;
    }
}